packages/devtools_app/lib/src/screens/memory/panes/leaks/leaks_pane.dart

```diff
@@ -148,122 +143,184 @@ class _LeaksPaneState extends State<LeaksPane>
+  }
+}
+
+class _LeaksPaneState extends State<LeaksPane>
+    with
+        AutoDisposeMixin,
+        ProvidedControllerMixin<MemoryController, LeaksPane> {
+  late _LeaksPaneController _controller;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    if (!initController()) return;
+    _controller = _LeaksPaneController(memoryController: controller);
+    cancelStreamSubscriptions();
+    _subscribeForMemoryLeaksMessages();
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
   void _subscribeForMemoryLeaksMessages() {
     autoDisposeStreamSubscription(
       serviceManager.service!.onExtensionEventWithHistory.listen((event) async {
-        if (event.extensionKind == _extensionKindToReceiveLeaksSummary) {
-          _receivedLeaksSummary(event);
+        if (event.extensionKind ==
+            _Constants.extensionKindToReceiveLeaksSummary) {
+          _controller.receivedLeaksSummary(event);
         }
       }),
     );
 
     autoDisposeStreamSubscription(
       serviceManager.service!.onExtensionEvent.listen((event) async {
-        if (event.extensionKind == _extensionKindToReceiveLeaksDetails) {
-          await _receivedLeaksDetails(event);
+        if (event.extensionKind ==
+            _Constants.extensionKindToReceiveLeaksDetails) {
+          await _controller.receivedLeaksDetails(event);
         }
       }),
     );
   }
 
-  Future<void> _requestLeaks() async {
-    _analysis.status.value = AnalysisStatus.Ongoing;
-    _analysis.message.value = 'Requested details from the application.';
-
-    await _invokeMemoryLeakTrackingExtension(
-      <String, dynamic>{
-        // TODO(polina-c): reference the constant in Flutter
-        // https://github.com/flutter/devtools/issues/3951
-        'requestDetails': 'true',
-      },
-    );
-  }
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder<bool>(
+      valueListenable: _controller.leakSummaryReceived,
+      builder: (_, leakSummaryReceived, __) {
+        if (!leakSummaryReceived) {
+          return Column(
+            children: const [
+              _InformationButton(),
+              Text('No information about memory leaks yet.'),
+            ],
+          );
+        }
 
-  Future<void> _forceGC() async {
-    _analysis.status.value = AnalysisStatus.Ongoing;
-    _analysis.message.value = 'Forcing full garbage collection...';
-    await _invokeMemoryLeakTrackingExtension(
-      <String, dynamic>{
-        // TODO(polina-c): reference the constant in Flutter
-        // https://github.com/flutter/devtools/issues/3951
-        'forceGC': 'true',
+        return Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            AnalysisStatusView(
+              controller: _controller.status,
+              analysisStarter: Row(
+                children: [
+                  const _InformationButton(),
```

Comment:
since this information button is to be shown in multiple views, maybe this should be the right aligned action on the header when the leaks tab is selected. Something like this:
![Screen Shot 2022-07-07 at 4 15 56 PM](https://user-images.githubusercontent.com/43759233/177886795-bb075f17-c608-441e-b1b9-b01b9d26fe62.png)

Reply:
This information is specific to leaks. I plan to link different doc for different panes, to make adoption easier for user.

Reply:
If I was user I would think the button is for page, not for tab.
Let's chat with Mindy about it.

Reply:
> To add some more context, here are four other tab views in DevTools where we dynamically change the right-aligned actions based on the selected tab. My main concern is making our UX consistent across DevTools. ![tabs-1](https://user-images.githubusercontent.com/43759233/178024533-3b3244e2-78ea-4c52-979d-c21d1927521f.gif) [ ](https://user-images.githubusercontent.com/43759233/178024533-3b3244e2-78ea-4c52-979d-c21d1927521f.gif) ![tabs-1](https://user-images.githubusercontent.com/43759233/178024533-3b3244e2-78ea-4c52-979d-c21d1927521f.gif) [ ](https://user-images.githubusercontent.com/43759233/178024533-3b3244e2-78ea-4c52-979d-c21d1927521f.gif) ![tabs-2](https://user-images.githubusercontent.com/43759233/178024559-46f8033d-84c1-4a39-a614-bf10ee6d6684.gif) [ ](https://user-images.githubusercontent.com/43759233/178024559-46f8033d-84c1-4a39-a614-bf10ee6d6684.gif) ![tabs-2](https://user-images.githubusercontent.com/43759233/178024559-46f8033d-84c1-4a39-a614-bf10ee6d6684.gif) [ ](https://user-images.githubusercontent.com/43759233/178024559-46f8033d-84c1-4a39-a614-bf10ee6d6684.gif) ![tabs-3](https://user-images.githubusercontent.com/43759233/178024567-d68f81a5-dbd2-49e0-a799-dbd136e731d8.gif) [ ](https://user-images.githubusercontent.com/43759233/178024567-d68f81a5-dbd2-49e0-a799-dbd136e731d8.gif) ![tabs-3](https://user-images.githubusercontent.com/43759233/178024567-d68f81a5-dbd2-49e0-a799-dbd136e731d8.gif) [ ](https://user-images.githubusercontent.com/43759233/178024567-d68f81a5-dbd2-49e0-a799-dbd136e731d8.gif) ![tabs-4](https://user-images.githubusercontent.com/43759233/178024574-c2b1f9c3-dba5-4c7d-a962-61804b2cdf46.gif) [ ](https://user-images.githubusercontent.com/43759233/178024574-c2b1f9c3-dba5-4c7d-a962-61804b2cdf46.gif) ![tabs-4](https://user-images.githubusercontent.com/43759233/178024574-c2b1f9c3-dba5-4c7d-a962-61804b2cdf46.gif) [ ](https://user-images.githubusercontent.com/43759233/178024574-c2b1f9c3-dba5-4c7d-a962-61804b2cdf46.gif)

In general, I agree that we should maintain consistency between screens as much as possible, however, we should make sure that if we are choosing to follow a design guideline for "consistency" that it is also serving our users.

One question I still have is how are you currently determining which options belong in the upper right part of the menu bar vs other parts of the window below the menu bar?

In this case, you could place the buttons in the spot that is the most "consistent" and then I could easily run a few quick usability tests to see if that matches users' mental models and gut check the rest of the design as well.

Reply:
> I agree that we should maintain consistency between screens as much as possible, however, we should make sure that if we are choosing to follow a design guideline for "consistency" that it is also serving our users.

+1.

> One question I still have is how are you currently determining which options belong in the upper right part of the menu bar vs other parts of the window below the menu bar?

In most cases, there aren't buttons below the menu bar. The content section is usually some sort of data visualization, and the right aligned buttons do something to this data.
![Untitled drawing](https://user-images.githubusercontent.com/43759233/178534472-a0308193-63d7-473d-beb7-00fea620e634.png)

> I could easily run a few quick usability tests to see if that matches users' mental models and gut check the rest of the design as well

sounds great!

Reply:
Let me share my concerns (I am not specialist in UX, so all of them may turn out to be unreal). 

It is known that upper-right corner of a website is the area that draws the least attention from the human eye (https://www.sitepoint.com/web-layout-design-making-the-most-of-the-top-right-corner/).

This means we should put here only buttons that user upfront knows should be here and will intentionally search for them (like search or menu or documentation about the entire screen).

I do not think user will search for documentation about concrete feature. So, if we want use to ever click (?), we need to put it very close the button, that executes the feature.

Again, I am not specialist in UX, my knowledge is very fragmental and I will appreciate professional guidance.